src/vcpkg/commands.add.cpp

```diff
@@ -41,8 +46,69 @@ namespace vcpkg::Commands
             {
                 Checks::exit_with_message(VCPKG_LINE_INFO, "add port requires an active manifest file.\n");
             }
+            const std::vector<ParsedQualifiedSpecifier> specs = Util::fmap(args.command_arguments, [&](auto&& arg) {
+                ExpectedS<ParsedQualifiedSpecifier> value = parse_qualified_specifier(std::string(arg));
+                if (value)
+                {
+                    if (!value.value_or_exit(VCPKG_LINE_INFO).triplet)
+                    {
+                        return value.value_or_exit(VCPKG_LINE_INFO);
+                    }
+                    print2(Color::error,
+                           "Error: Triplet expressions are not allowed here. Use the --triplet parameter when running "
+                           "`vcpkg install`.\n");
+                }
+                else
+                {
+                    print2(Color::error, value.error());
+                }
+                Checks::exit_fail(VCPKG_LINE_INFO);
+            });
+            const auto& manifest_path = paths.get_manifest_path().value_or_exit(VCPKG_LINE_INFO);
+            auto maybe_manifest_scf = SourceControlFile::parse_manifest_object(manifest_path, *manifest);
+            if (!maybe_manifest_scf)
+            {
+                print_error_message(maybe_manifest_scf.error());
+                print2("See ", docs::manifests_url, " for more information.\n");
+                Checks::exit_fail(VCPKG_LINE_INFO);
+            }
 
-            Checks::exit_with_message(VCPKG_LINE_INFO, "add port will be added in a future vcpkg release.\n");
+            auto& manifest_scf = *maybe_manifest_scf.value_or_exit(VCPKG_LINE_INFO);
+            for (const auto& spec : specs)
+            {
+                auto dep = Util::find_if(manifest_scf.core_paragraph->dependencies, [&spec](Dependency& dep) {
+                    return dep.name == spec.name && !dep.host &&
+                           structurally_equal(spec.platform.value_or(PlatformExpression::Expr()), dep.platform);
+                });
+                if (dep == manifest_scf.core_paragraph->dependencies.end())
+                {
+                    manifest_scf.core_paragraph->dependencies.push_back(
+                        Dependency{spec.name, spec.features.value_or({}), spec.platform.value_or({})});
+                }
+                else
+                {
+                    if (spec.features)
+                    {
```

Comment:
```suggestion
                else if (spec.features)
                {
```

this is just simpler and requires less rightward drift.